File: app/src/main/java/io/lerk/lrkFM/LrkFMApp.java
package io.lerk.lrkFM;

import android.app.Application;
import android.content.Context;

import java.lang.ref.WeakReference;

/**
 * Main application class.
 *
 * @author Lukas Fülling (lukas@k40s.net)
 */
public class LrkFMApp extends Application {

    public static final String CHANNEL_ID = "lrkFM";
    private static WeakReference<Context> context;

    @Override
    public void onCreate() {
        super.onCreate();
        context = new WeakReference<>(this);
    }

    public static Context getContext() {
        return context.get();
    }


}

File: app/src/main/java/io/lerk/lrkFM/LrkFMApp.kt
package io.lerk.lrkFM

import android.app.Application
import android.content.Context
import java.lang.ref.WeakReference

/**
 * Main application class.
 *
 * @author Lukas Fülling (lukas@k40s.net)
 */
class LrkFMApp : Application() {
    override fun onCreate() {
        super.onCreate()
        context = WeakReference(this)
    }

    companion object {
        const val CHANNEL_ID = "lrkFM"
        private var context: WeakReference<Context>? = null
        fun getContext(): Context? {
            return context!!.get()
        }
    }
}
